test/common/crudtester.go

```diff
@@ -714,27 +715,133 @@ func (c *FederatedTypeCrudTester) versionForCluster(version *fedv1a1.PropagatedV
 	return ""
 }
 
+func (c *FederatedTypeCrudTester) CheckRemoteStatus(fedObject *unstructured.Unstructured, targetObject *unstructured.Unstructured) {
+	for clusterName := range c.testClusters {
+		clusterConfig := c.testClusters[clusterName].Config
+
+		kubeClient := kubeclientset.NewForConfigOrDie(clusterConfig)
+		WaitForNamespaceOrDie(c.tl, kubeClient, clusterName, targetObject.GetNamespace(),
+			c.waitInterval, 30*time.Second)
+
+		util.AddManagedLabel(targetObject)
+		labeledObj, err := CreateResource(clusterConfig, c.typeConfig.GetTargetType(), targetObject)
+		if err != nil {
+			c.tl.Fatalf("Failed to create labeled resource in cluster %q: %v", clusterName, err)
+		}
+
+		clusterClient := genericclient.NewForConfigOrDie(clusterConfig)
+		defer func() {
+			err := clusterClient.Delete(context.TODO(), labeledObj, labeledObj.GetNamespace(), labeledObj.GetName())
+			if err != nil {
+				c.tl.Fatalf("Unexpected error: %v", err)
+			}
+		}()
+
+		c.tl.Log("Checking that the resource has status")
+		var objStatus interface{}
+		err = wait.PollImmediate(c.waitInterval, wait.ForeverTestTimeout, func() (bool, error) {
+			obj := &unstructured.Unstructured{}
+			obj.SetGroupVersionKind(labeledObj.GroupVersionKind())
+			err := clusterClient.Get(context.TODO(), obj, labeledObj.GetNamespace(), labeledObj.GetName())
+			if err != nil {
+				c.tl.Errorf("Error retrieving kubefed cluster object resource: %v", err)
+				return false, nil
+			}
+
+			objStatus = obj.Object[util.StatusField]
+			c.tl.Logf("Show kubefed cluster object status: %v", objStatus)
+			return (obj.Object[util.StatusField] != nil), nil
+		})
+		if err != nil {
+			c.tl.Fatal("Timed out waiting for the resource to have a status field")
+		}
+		c.tl.Logf("Kubefed cluster object status: %v", objStatus)
+
+		c.tl.Log("Checking that the federated resource has a remote status field")
+		objRemoteStatus, err := c.getRemoteStatus(fedObject, clusterName)
+		if err != nil {
+			c.tl.Fatal("Timed out waiting for the federated resource to have a remote status field")
+		}
+		if objRemoteStatus == nil {
+			c.tl.Fatal("Federated object remote status is empty")
+		}
+		c.tl.Logf("Show federated object remote status %v", objRemoteStatus)
+	}
+}
+
+func (c *FederatedTypeCrudTester) getRemoteStatus(fedObject *unstructured.Unstructured, clusterName string) (interface{}, error) {
+	apiResource := c.typeConfig.GetFederatedType()
+	qualifiedName := util.NewQualifiedName(fedObject)
+
+	client := c.resourceClient(apiResource)
+	var remoteStatusObj interface{}
+	// The default is normally 30 seconds
+	waitTimeout := 6 * wait.ForeverTestTimeout
+	err := wait.PollImmediate(c.waitInterval, waitTimeout, func() (bool, error) {
+		fedObj, err := client.Resources(qualifiedName.Namespace).Get(context.Background(), qualifiedName.Name, metav1.GetOptions{})
+		if err != nil {
+			c.tl.Errorf("An unexpected error occurred while polling for remote status: %v", err)
+			return false, nil
+		}
+
+		if err == nil {
```

Comment:
this is always true and is unnecessary.

if err != nil the function returns... at this point... err is nil

Reply:
which will remove 1 of the returns and should just be `return false, nil`